### ISSUE TITLE
uninstall cannot be used (without --kill) to merely de-init sub-datasets

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#1079]
The best achievable result on a freshly installed (without any data or local changes) sub-dataset is a total removal, including from the history of the super-dataset:

```shell

(git)hopa:/tmp/datasets.datalad.org/crcns[master]git
$> datalad uninstall aa-1                                              
[ERROR  ] will not uninstall content in subdatasets without the recursive flag [uninstall.py:__call__:229] (ValueError) 

$> datalad uninstall --recursive aa-1
Nothing was uninstalled

$> datalad uninstall --remove-handles aa-1            
[ERROR  ] will not uninstall content in subdatasets without the recursive flag [uninstall.py:__call__:229] (ValueError) 
(venv-tests) 2 16864 ->1 [1].....................................:Tue 01 Nov 2016 04:37:49 PM EDT:.
(git)hopa:/tmp/datasets.datalad.org/crcns[master]git
$> datalad uninstall --remove-handles --recursive aa-1
[ERROR  ] will not remove the entire dataset (with history) unless forced [uninstall.py:_uninstall:70] (ValueError) 

$> datalad uninstall --remove-handles --recursive --remove-history aa-1
1 item was uninstalled:
Dataset: /tmp/datasets.datalad.org/crcns/aa-1

$> git show
commit 468ab621d78c88d9898ad29d56dad69e7eb52ec1
Author: Yaroslav Halchenko <debian@onerussian.com>
Date:   Tue Nov 1 16:38:22 2016 -0400

    [DATALAD] uninstalled content

diff --git a/.gitmodules b/.gitmodules
index db3486a..e1d3b21 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -112,9 +112,6 @@
 [submodule "bst-1"]
        path = bst-1
        url = ./bst-1
-[submodule "aa-1"]
-   path = aa-1
-   url = ./aa-1
 [submodule "aa-2"]
        path = aa-2
        url = ./aa-2
@@ -154,5 +151,3 @@
 [submodule "fcx-1"]
        path = fcx-1
        url = ./fcx-1
-
-
diff --git a/aa-1 b/aa-1
deleted file mode 160000
index b923800..0000000
--- a/aa-1
+++ /dev/null
@@ -1 +0,0 @@
-Subproject commit b923800c386837c92bc3d440070ee219606c3ce8
(venv-tests) 2 16867 [1].....................................:Tue 01 Nov 2016 04:38:56 PM EDT:.
(git)hopa:/tmp/datasets.datalad.org/crcns[master]git
$> 
```

**Comment by mih** [comment]
Observations that may be useful for work towards a fix:

````
$ datalad ls -r
.            [annex]  master  - 2016-11-11/08:07:20  OK
src/dicoms   [annex]  master  - 2016-11-11/07:57:59  OK
# needs at least the following to get rid of the submodule
$ datalad uninstall -d src/dicoms  --remove-handles --remove-history
1 item was uninstalled:
...
$ git status
On branch master
Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	deleted:    src/dicoms

no changes added to commit (use "git add" and/or "git commit -a")
$ datalad save src/dicoms
$ git status
On branch master
nothing to commit, working directory clean
$ git submodule
-e248ddca0e42bbce19440d6075f64ec144986f07 src/dicoms
````

So it is possible, but convoluted.

**Comment by mih** [comment]
FTR: `uninstall` correctly complains fails when some annexed files do not have enough redundant copies, but I didn't check whether it would also complain about the current branch not being pushed to any remote.

**Comment by mih** [comment]
More logs:

````
$ datalad ls -r
.                    [annex]  master  - 2016-11-11/08:41:54  OK
src/raw              [annex]  master  - 2016-09-02/09:38:19  OK
src/raw/src/dicoms   not installed
$ datalad uninstall --remove-handles --remove-history src/raw -r
[ERROR  ] Cannot delete module at /src/raw as there are new commits [base.py:remove:843] (InvalidGitRepositoryError) 
````

this is absolutely undesired (and a variant of what Yarik posted). Here is what is going on:

````
$ git -C src/raw diff HEAD~1
diff --git a/.gitmodules b/.gitmodules
index e01de9a..e69de29 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -1,3 +0,0 @@
-[submodule "src/dicoms"]
-       path = src/dicoms
-       url = ../dicoms/
diff --git a/src/dicoms b/src/dicoms
deleted file mode 160000
index 0d4592f..0000000
--- a/src/dicoms
+++ /dev/null
@@ -1 +0,0 @@
-Subproject commit 0d4592f2b78014a94b1211d8e0761bf24f9146c8
````

So it unregistered the uninstalled submodules -- not kosher.

**Comment by mih** [comment]
Fixed by #1130
